mlir/include/mlir/Dialect/MIOpen/utility/loweringUtils.h

```diff
@@ -22,62 +22,27 @@
 namespace mlir {
 namespace miopen {
 
-// Heuristic logic to compute KBlock for backward weight atomic add kernel.
-// The logic is adopted from MIOpen.
-//
-// The logic searches within the range of [1, 20 * number of CUs / gridSize],
-// where gridSize is the original number of workgroups required for the
-// convolution, and find the largest KBlock number which preserves the 2
-// contraints:
-// - GemmK (before splitting) = KBlock * KPerBlock * KPack * GemmK (after
-// splitting).
-// - n (batch size) is divisible by KBlock.
-//
-// 20 is a magic number obtained in MIOpen after empirical testing. It offers a
-// reasonable reduction of GemmK after splitting, without incurring too much
-// overheads on atomic adds. One potential future work is to make this value be
-// tunable.
-inline int64_t calculateKBlockNum(int64_t n, int64_t ho, int64_t wo, int64_t g,
-                                  int64_t k, int64_t c, int64_t y, int64_t x,
-                                  int64_t MPerBlock, int64_t NPerBlock,
-                                  int64_t KPerBlock, int64_t KPack,
-                                  int64_t num_cu) {
-  const int64_t KPerGroup = k / g;
-  const int64_t CPerGroup = c / g;
-  const int64_t gemmM = KPerGroup;
-  const int64_t gemmN = CPerGroup * y * x;
-
-  const int64_t gemmK = n * ho * wo;
-  int64_t gemmKBlock = 1;
-
-  assert(gemmM % MPerBlock == 0);
-  assert(gemmN % NPerBlock == 0);
-  assert(gemmK % (KPerBlock * KPack) == 0);
```

Comment:
Looking at the nightly failure it looks like we need to reject parameters that doesn't satisfy this.
  - Line 53 and 54 are okay because there are already logic rejecting parameters that doesn't satisfy

Reply:
I've already devised a fix. Working on proving it mathematically before submitting the PR.